src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { V1beta1DataVolumeSpec } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { Form } from '@patternfly/react-core';
+
+import {
+  buildFields,
+  extractParameterNameFromMetadataName,
+  getTemplateStorageQuantity,
+} from '../../utils';
+import { ExpandableCustomizeSourceSection } from '../CustomizeSource/ExpandableCustomizeSourceSection';
+import { ExpandableOptionsFields } from '../ExpandableOptionalFields';
+import { FieldGroup } from '../FieldGroup';
+import { FormActionGroup } from '../FormActionGroup';
+
+import { useCustomizeFormSubmit } from './useCustomizeFormSubmit';
+
+type CustomizeFormWithDiskProps = {
+  template: V1Template;
+};
+
+export const CustomizeFormWithDisk: React.FC<CustomizeFormWithDiskProps> = ({ template }) => {
+  const { t } = useKubevirtTranslation();
+  const [customDiskSource, setCustomDiskSource] = React.useState<V1beta1DataVolumeSpec>();
+  const [onSubmit, loaded, error] = useCustomizeFormSubmit(template, customDiskSource);
+
+  const parameterForName = extractParameterNameFromMetadataName(template);
+
+  const [requiredFields, optionalFields] = buildFields(template, [parameterForName], t);
+
+  const onDiskSourceChange = React.useCallback((newDiskSource) => {
+    setCustomDiskSource(newDiskSource);
+  }, []);
+
+  return (
+    <Form onSubmit={onSubmit}>
+      {requiredFields?.map((field) => (
```

Comment:
name override inputs are above strorage section
required fields are below the storage section

---
order of inputs is:

name
storage (disk or cd)
add drivers disk check box
required parameters
optional parameters

Reply:
really? So strange

Reply:
fixed now